packages/smart-vaults/src/utils/LightSyncMultiSigner.sol

```diff
@@ -84,56 +57,53 @@ abstract contract LightSyncMultiSigner is MultiSigner {
     /// @dev reverts if validation fails.
     function _validateSignerSetUpdate(
         MultiSignerLib.MultiSignerStorage storage $_,
-        SignerSetUpdate memory signerUpdate_,
+        SignerSetUpdate memory signerSetUpdate_,
         uint256 nonce_
     )
         internal
         view
     {
         if (
             !MultiSignerSignatureLib.isValidSignature(
-                $_, _getSignerUpdateHash(signerUpdate_, nonce_), signerUpdate_.normalSignature
+                $_, _getSignerUpdateHash(signerSetUpdate_, nonce_), signerSetUpdate_.normalSignature
             )
-        ) revert SignerSetUpdateValidationFailed(signerUpdate_);
+        ) revert SignerSetUpdateValidationFailed(signerSetUpdate_);
     }
 
-    function _processSignerSetUpdate(SignerSetUpdate memory signerUpdate_) internal {
-        SignerUpdateParam[] memory updateParams = signerUpdate_.updateParams;
-        uint256 numUpdates = updateParams.length;
-        for (uint256 i; i < numUpdates; i++) {
-            _processSignerUpdateParam(updateParams[i]);
-        }
+    function _processSignerSetUpdate(SignerSetUpdate memory signerSetUpdate_) internal {
+        (bytes memory signer, uint8 index) = abi.decode(signerSetUpdate_.data, (bytes, uint8));
+        _addSigner(signer, index);
```

Comment:
does this reverting for idx collision cause an issue for dummy sig gas estimation?

Reply:
no, this would be a bad light sync update which should not go through, and we would have the right signature for this to go through as long as the update is valid